caps/eth.md

```diff
@@ -105,25 +105,185 @@ of it (either because it was previously sent or because it was informed from thi
 originally). This is usually achieved by remembering a set of transaction hashes recently
 relayed by the peer.
 
-Transactions must be validated before re-propagating them. Relaying an invalid transaction
-results in peer disconnection.
+### Transaction Encoding and Validity
+
+Transaction objects exchanged by peers have one of two encodings. In definitions across
+this specification, we refer to transactions of either encoding using the identifier `txₙ`.
+
+    tx = {legacy-tx, typed-tx}
+
+Untyped, legacy transactions are given as an RLP list.
+
+    legacy-tx = [
+        nonce: P,
+        gas-price: P,
+        gas-limit: P,
+        recipient: {B_0, B_20},
+        value: P,
+        data: B,
+        V: P,
+        R: P,
+        S: P,
+    ]
+
+[EIP-2718] typed transactions are encoded as RLP byte arrays where the first byte is the
+transaction type (`tx-type`) and the remaining bytes are opaque type-specific data.
+
+    typed-tx = rlp(tx-type || tx-data)
+
+Transactions must be validated when they are received. Validity depends on the Ethereum
+chain state. The specific kind of validity this specification is concerned with is not
+whether the transaction can be executed successfully by the EVM, but only whether it is
+acceptable for temporary storage in the local pool and for exchange with other peers.
+
+Transactions must be validated according to the rules below. While the encoding of typed
+transactions is opaque, it is assumed that their `tx-data` provides values for `nonce`,
+`gas-price`, `gas-limit`, and that the sender account of the transaction can be determined
+from their signature.
+
+- If the transaction is typed, the `tx-type` must be known to the implementation. Defined
+  transaction types may be considered valid even before they become acceptable for
+  inclusion in a block. Implementations should disconnect peers sending transactions of
+  unknown type.
+- The signature (`V`, `R`, `S` values) must be valid according to the signature schemes
+  supported by the chain. For typed transactions, signature handling is defined by the EIP
+  introducing the type. For legacy transactions, the two schemes in active use are the
+  basic 'Homestead' scheme and the [EIP-155] scheme.
```

Comment:
As of Berlin, not all transactions have a `V`.  2930 transactions have `yParity, r, s`.

Reply:
`V`, `R`, `S` removed